### PR TITLE
feat(sdk): add timeout handler to simulator queue

### LINKED ISSUE #165: Simulate queue timeout property
There is a "timeout" property that can be passed to the props of a `Queue` resource, but it has no affect on the current simulation.

A timeout of 5s means that if a consumer processes a message from a queue and takes more than 5 seconds before returning a successful response, then the response should get ignored and the message is added back to the queue.

### PR DESCRIPTION
**Pull request by fynnfluegge**
Closes #165.

Console crash mentioned in #1213 should also be fixed with this PR, but I let this issue open because `retentionPeriod` still needs to be added to terraform queue implementations.

To be consistent I also changed the default timeout period in the simulator queue to 10s, since `@default Duration.fromSeconds(10)` is documented in base class `cloud/queue.ts`.

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: libs/wingsdk/src/target-sim/queue.inflight.ts [hunk 1]
```diff
@@ -102,12 +104,27 @@ export class Queue implements IQueueClient, ISimulatorResourceInstance {
...
           });
-          this.messages.push(...messages);
+          this.pushMessagesBackToQueue(messages);
         });
         processedMessages = true;
       }
     } while (processedMessages);
   }
+
+  public async pushMessagesBackToQueue(messages: any): Promise<void> {
```
Comment (Chriscbr) [inline libs/wingsdk/src/target-sim/queue.inflight.ts:114]: Can we add a more specific type for `messages`?

File: libs/wingsdk/src/target-sim/queue.inflight.ts [hunk 2]
```diff
@@ -102,12 +104,27 @@ export class Queue implements IQueueClient, ISimulatorResourceInstance {
...
+      this.context.addTrace({
+        data: {
+          message: `${messages.length} messages pushed back to queue after timeout.`,
+        },
+        sourcePath: this.context.resourcePath,
+        sourceType: QUEUE_TYPE,
+        type: TraceType.RESOURCE,
+        timestamp: new Date().toISOString(),
+      });
+    }, this.timeout * 1000);
```
Comment (Chriscbr) [inline libs/wingsdk/src/target-sim/queue.inflight.ts:126]: If a function takes 5 seconds to process a message and fail, and the queue's timeout is set for 10 seconds, should we still wait the full 10 seconds before reprocessing it, or should we only wait for the "remaining" time of 5 seconds, so to speak? (I haven't played with this in AWS SQS so I'm not sure off the top of my head how it works there, but figured I'd ask)

**Review by Chriscbr — COMMENTED** [review]
Wow, looks great (also thanks for the small fixes along the way)! I had one small question, but otherwise I think everything looks good.

File: libs/wingsdk/src/target-sim/queue.inflight.ts (hunk 2 above)
Comment (fynnfluegge) [reply libs/wingsdk/src/target-sim/queue.inflight.ts:136]: Good point, I was also thinking about this 😅
 Usually the behaviour is, that the subscriber will delete the message after successfull processing it. During the timeout period, after a subscriber fetched the message from the queue, it remains in the queue but cannot be requested from other subscribers. [This](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-visibility-timeout.html) explanation is really good. To 100% simulate this behaviour the simulator queue might need a rework. This is not trivial I think.
But yes, my fix would currently have a timeout period of 15s if the subscriber throws an error after 5s and and the timeout is 10s. Is it fine or should we simulate this more precise, what do you think?

File: libs/wingsdk/src/target-sim/queue.inflight.ts (hunk 1 above)
Comment (fynnfluegge) [reply libs/wingsdk/src/target-sim/queue.inflight.ts]: Sure ✅

**Review by Chriscbr — APPROVED** [review]
🙏

File: libs/wingsdk/src/target-sim/queue.inflight.ts (hunk 2 above)
Comment (fynnfluegge) [reply libs/wingsdk/src/target-sim/queue.inflight.ts:136]: I agree, it should be more accurate in the future, I will create a follow-up feature issue for this ✅

**Pull request merged**